app/src/main/java/org/systers/mentorship/view/fragments/ProfileFragment.kt

```diff
@@ -37,34 +54,149 @@ class ProfileFragment : BaseFragment() {
         super.onActivityCreated(savedInstanceState)
 
         setHasOptionsMenu(true)
+        baseActivity.setSupportActionBar(tbProfile)
+        originalEditTextDrawable = etUserName.background
+
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
+            switchAvailableToMentor.buttonTintList =
+                ColorStateList.valueOf(ContextCompat.getColor(context!!, R.color.colorPrimary))
+        }
 
         profileViewModel = ViewModelProviders.of(activity!!).get(ProfileViewModel::class.java)
-        profileViewModel.successfulGet.observe(this, Observer {
-            successful ->
+        profileViewModel.successfulGet.observe(this, Observer { successful ->
             baseActivity.hideProgressDialog()
             if (successful != null) {
                 if (successful) {
                     fragmentProfileBinding.user = profileViewModel.user
                 } else {
-                    Snackbar.make(fragmentProfileBinding.root, profileViewModel.message,
-                            Snackbar.LENGTH_LONG).show()
+                    Snackbar.make(
+                        fragmentProfileBinding.root, profileViewModel.message,
+                        Snackbar.LENGTH_LONG
+                    ).show()
                 }
             }
         })
+
+        profileViewModel.editInProgress.observe(this, Observer { editing ->
+            activity?.invalidateOptionsMenu() // Refresh options on toolbar
+
+            val inputs: List<View> = arrayListOf(
+                etUserName,
+                tiEtBio,
+                tiEtSlackUsername,
+                tiEtLocation,
+                tiEtOccupation,
+                tiEtOrganization,
+                tiEtSkills,
+                tiEtInterests
+            )
+
+            val switches: List<View> = arrayListOf(switchAvailableToMentor, switchNeedMentorship)
+
+            inputs.forEach {
+                it.isEnabled = editing
+                it.background = if (editing) {
+                    originalEditTextDrawable
+                } else null
+            }
+
+            switches.forEach {
+                it.isEnabled = editing
```

Comment:
I think it will be great to deeper your knowledge about data binding :) not everything we have to do in code. Android is a great place where we have many concept involved and we can get same effect with in many different coding principals and patterns. Great work!

Reply:
Yeah, you're right. This can be done better way using DataBinding. I'm gonna fix it.

Reply:
Fantastic!